src/include/optimizer/rule_rewrite.h

```diff
@@ -0,0 +1,45 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// rule_rewrite.h
+//
+// Identification: src/include/optimizer/rule_rewrite.h
+//
+// Copyright (c) 2015-16, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "optimizer/rule.h"
+#include "optimizer/absexpr_expression.h"
+
+#include <memory>
+
+namespace peloton {
+namespace optimizer {
+
+/* Rules are applied from high to low priority */
+enum class RulePriority : int {
```

Comment:
(I know nothing about optimizers) Is priority a well-defined concept for optimization or is this a rough heuristic? What happens if you have two rules of the same priority that can both be applied, is it always arbitrary which should go first?